src/eval/tests/trie.rs

```diff
@@ -10,34 +10,29 @@ fn trie_lang() {
 
     install(s, &mut lang);
 
-    let expr = "(let ((trie (.lurk.trie.new)))
+    // TODO wrap .lurk.trie.new, .lurk.trie.lookup and .lurk.trie.insert in lambdas
+    let expr = "(let ((trie .lurk.trie.new))
                       trie)";
     let res = s
         .read("0x1cc5b90039db85fd519af975afa1de9d2b92960a585a546637b653b115bc3b53")
         .unwrap();
 
     test_aux(s, &expr, Some(res), None, None, None, 3, Some(&lang));
 
-    // TODO: Coprocessors need to evaluate their arguments for this to work.
-    //       See https://github.com/lurk-lab/lurk-rs/issues/398.
-    // let expr2 = "(let ((trie (.lurk.trie.new))
-    //                    (found (.lurk.trie.lookup trie 123)))
-    //                   found)";
-
-    let expr2 = "(.lurk.trie.lookup 0x1cc5b90039db85fd519af975afa1de9d2b92960a585a546637b653b115bc3b53 123)";
+    let expr2 = "((lambda (_x _y) .lurk.trie.lookup) .lurk.trie.new 123)";
     let res2 = s.intern_opaque_comm(Fr::zero());
 
-    test_aux(s, &expr2, Some(res2), None, None, None, 1, Some(&lang));
+    test_aux(s, &expr2, Some(res2), None, None, None, 9, Some(&lang));
 
-    let expr3 = "(.lurk.trie.insert 0x1cc5b90039db85fd519af975afa1de9d2b92960a585a546637b653b115bc3b53 123 456)";
+    let expr3 = "((lambda (_x _y _z) .lurk.trie.insert) .lurk.trie.new 123 456)";
```

Comment:
This is fine as an intermediate example of what the coprocessor 'macroexpansion' should potentially look like. I think when we are done, this should revert back to the code I would like to have written in the first place for the trie test:
```
(let ((trie (.lurk.trie.new))
  (.lurk.trie.insert trie 123 456))
```